classes/local/form/request_evaluation_form.php

```diff
@@ -0,0 +1,305 @@
+
+global $CFG;
+require_once($CFG->libdir . '/formslib.php');
+require_once($CFG->dirroot . '/local/lsf_unification/lib_his.php');
+
+/**
+ * Evaluation request form
+ *
+ * @package    block_evasys_sync
+ * @copyright  2022 Justus Dieckmann WWU
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class request_evaluation_form extends moodleform {
+
+    private $course;
+    private $lsfcourses;
+    private $defaulttimeframe;
+    private $defaultdata;
+
+    public function __construct($course, $action = null) {
+        $this->course = $course;
+        $this->init();
+        parent::__construct($action, null, 'post', '', ['class' => 'evasys-request-mform']);
+    }
+
+    private function init() {
+        global $USER, $PAGE;
+        establish_secondary_DB_connection();
+        $this->lsfcourses = [];
+        foreach (get_teachers_course_list($USER->username) as $v) {
+            $this->lsfcourses[$v->veranstid] = $v->info;
+        }
+        close_secondary_DB_connection();
+        $this->defaulttimeframe = evasys_synchronizer::get_standard_timemode($this->course->category);
+        $this->defaultdata = $this->build_expanded_default_data(null);
+        $PAGE->requires->js_call_amd('block_evasys_sync/evaluation_request', 'init');
+    }
+
+    public function display()
+    {
+        parent::display();
+
+        // Create template for date time selector for JS. This way instead of js argument in order to get rid of...
+        // "Too much data"-warning. This is not nice, but if this is what it takes to have no warning...
+        $a = new \MoodleQuickForm_date_time_selector('myevasyselementname', 'MyEvasysElementLabel', [], []);
+        $a->setMoodleForm(new \MoodleQuickForm('', '', ''));
+        $a->_createElements();
+        $renderer = new \MoodleQuickForm_Renderer();
+        $a->accept($renderer);
+        echo '<div id="evasys-dateselectortemplate" hidden>';
+        echo $renderer->_html;
+        echo '</div>';
+    }
+
+    /**
+     * Defines form elements.
+     */
+    protected function definition() {
+        $mform = $this->_form;
+
+        $mform->addElement('checkbox', 'donotevaluate', get_string('dont_evaluate_course', 'block_evasys_sync'));
+
+        $mform->addElement('html', '<fieldset id="fitem_id_sectionone" class="fitem grouponly" name="sectionone">');
+        $mform->disabledIf('sectionone', 'donotevaluate', 'checked');
+        $mform->disabledIf('sectiontwo', 'donotevaluate', 'checked');
+
+        $mform->addElement('checkbox', 'additionalveranstaltungen',
+            get_string('eval_additional_lsfcourses', 'block_evasys_sync'));
+
+        $mform->addElement('autocomplete', 'lsfcourses',
+            get_string('course_units', 'block_evasys_sync'),
+            $this->lsfcourses, ['multiple' => true]);
+
+        $mform->hideIf('lsfcourses', 'additionalveranstaltungen', 'notchecked');
+
+        $mform->addElement('checkbox', 'additionalcourses',
```

Comment:
This option did not work for me, the corresponding emails were not synchronized, even from the original course